calico_containers/tests/st/test_duplicate_ips.py

```diff
@@ -17,44 +17,39 @@ def test_duplicate_ips(self):
         host3 = DockerHost('host3')
 
         # Set up three workloads on three hosts
-        host1.execute("docker run -e CALICO_IP=192.168.1.1 --name workload1 -tid busybox",
-                      use_powerstrip=True)
-        host2.execute("docker run -e CALICO_IP=192.168.1.2 --name workload2 -tid busybox",
-                      use_powerstrip=True)
-        host3.execute("docker run -e CALICO_IP=192.168.1.3 --name workload3 -tid busybox",
-                      use_powerstrip=True)
+        workload1 = host1.create_workload("workload1", "192.168.1.1")
+        workload2 = host2.create_workload("workload2", "192.168.1.2")
+        workload3 = host3.create_workload("workload3", "192.168.1.3")
 
         # Set up the workloads with duplicate IPs
-        host1.execute("docker run -e CALICO_IP=192.168.1.4 --name dup1 -tid busybox",
-                      use_powerstrip=True)
-        host2.execute("docker run -e CALICO_IP=192.168.1.4 --name dup2 -tid busybox",
-                      use_powerstrip=True)
+        dup_ip = "192.168.1.4"
+        dup1 = host1.create_workload("dup1", dup_ip)
+        dup2 = host2.create_workload("dup2", dup_ip)
 
         host1.calicoctl("profile add TEST_PROFILE")
 
         # Add everyone to the same profile
-        host1.calicoctl("profile TEST_PROFILE member add workload1")
-        host1.calicoctl("profile TEST_PROFILE member add dup1")
-        host2.calicoctl("profile TEST_PROFILE member add workload2")
-        host2.calicoctl("profile TEST_PROFILE member add dup2")
-        host3.calicoctl("profile TEST_PROFILE member add workload3")
-
-        # Wait for the workload networking to converge.
-        ping = partial(host1.execute, "docker exec workload1 ping -c 4 192.168.1.4")
-        retry_until_success(ping, ex_class=ErrorReturnCode_1)
+        host1.calicoctl("profile TEST_PROFILE member add %s" % workload1)
+        host1.calicoctl("profile TEST_PROFILE member add %s" % dup1)
+        host2.calicoctl("profile TEST_PROFILE member add %s" % workload2)
+        host2.calicoctl("profile TEST_PROFILE member add %s" % dup2)
+        host3.calicoctl("profile TEST_PROFILE member add %s" % workload3)
 
         # Check for standard connectivity
-        host1.execute("docker exec workload1 ping -c 4 192.168.1.4")
-        host2.execute("docker exec workload2 ping -c 4 192.168.1.4")
-        host3.execute("docker exec workload3 ping -c 4 192.168.1.4")
+        ping = partial(workload1.ping, dup_ip)
```

Comment:
I get what you're saying, but in an ST environment a ping is (pretty much?) always a test, not an action to take where the side effect is import.

Executing the ping command already acts like a test in that it throws an exception if it fails.

If the difference is in retries (and you think that distinction is important enough to keep) then you need to name the methods so that difference is clear.

I don't think that the plausibility of other differences is reason for us to take action now.  We can always change things later if it becomes necessary.